bot-twitter/src/keeper.ts

```diff
@@ -0,0 +1,81 @@
+import { AuctionInitialInfo } from 'auctions-core/src/types';
+import getSigner from 'auctions-core/src/signer';
+import { bidOnTheAuction, enrichAuction } from 'auctions-core/src/auctions';
+import {
+    authorizeCollateral,
+    authorizeWallet,
+    getCollateralAuthorizationStatus,
+    getWalletAuthorizationStatus,
+} from 'auctions-core/src/authorizations';
+import { ETHEREUM_NETWORK, KEEPER_MIN_PROFIT_DAI } from './variables';
+
+async function participate(auction: AuctionInitialInfo) {
+    if (KEEPER_MIN_PROFIT_DAI === undefined) {
+        return;
+    }
+    const signer = await getSigner(ETHEREUM_NETWORK);
```

Comment:
As far as I understand, right now `await getSigner` will throw X errors in a raw every 30 seconds if `KEEPER_WALLET_PRIVATE_KEY` was not set. We want to avoid that and instead show `warn` message only during setup.

Therefore, you may also want to split `participate` into at least two parts:
- `participationSetup` where you validate variables, create signer and if anything goes wrong, logs error/warning accordingly
- something like `participateInAuction` where you can silently fail like it is now due to any validation problem from above. See twitter for inspiration on how it is structured there